edx_exams/apps/lti/views.py

```diff
@@ -13,31 +13,53 @@
 from lti_consumer.api import get_end_assessment_return, get_lti_1p3_launch_start_url
 from lti_consumer.data import Lti1p3LaunchData, Lti1p3ProctoringLaunchData
 from lti_consumer.models import LtiConfiguration
-from rest_framework.decorators import authentication_classes, permission_classes
+from rest_framework import status
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
 
+from edx_exams.apps.core.api import get_attempt_by_id, update_attempt_status
 from edx_exams.apps.core.models import ExamAttempt
+from edx_exams.apps.core.statuses import ExamAttemptStatus
 from edx_exams.apps.lti.utils import get_lti_root
 
 EDX_OAUTH_BACKEND = 'auth_backends.backends.EdXOAuth2'
 
 
+@api_view(['GET'])
```

Comment:
The start_proctoring view only handles GET requests. This doesn't feel as appropriate given that it changes the ExamAttempt state - it's a side effecting GET. However, the URL doesn't really uniquely define a resource (i.e. the ExamAttempt). A PUT would be more appropriate, but it would make it more difficult to manually test without wiring up the learning MFE to edx-exams because you can no longer just put the URL in the browser. What do you think? I can log a ticket to update the request handling once we have a better way to test the entire LTI flow via the learning MFE.

Reply:
I think this has to be a GET because it isn't really an 'API request' and instead a new webpage the browser links to. Not sure how to make the side effect here more elegant 🤔

Reply:
maybe we make sure it is idempotent-ish so in case the learner hits the same launch URL twice it doesn't error, it simply always results in download_software_clicked? (if that is a valid transition)

Reply:
Your second thought about the auto-submitting form is separate from the GET v. POST topic, right? I'm not seeing the connection. What do you mean by "more correct"? The specification does say a redirect is acceptable, so I'm trying to understand the benefit/difference.

Reply:
When I've done multiple launch in succession, I haven't had any errors in the state transitions, so it appears idempotent.

Reply:
I was thinking 'more correct' in terms of HTTP specification so that our launch url becomes a POST. But, for now at least I see no reason that extra complexity is worth it.

Reply:
Ah, I see. I thought you meant replace the LTI redirect here with an auto-posting form, which is what we do in the LMS, actually. That's where my confusion was.